wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InductionSlotBookingFollowUpCron.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+use Civi\Api4\Activity;
+use Civi\Api4\MessageTemplate;
+
+/**
+ * Goonjcustom.InductionSlotBookingFollowUp API specification (optional)
+ * This is used for documentation and validation.
+ *
+ * @param array $spec
+ *   description of fields supported by this API call.
+ *
+ * @return void
+ */
+function _civicrm_api3_goonjcustom_induction_slot_booking_follow_up_cron_spec(&$spec) {
+  // There are no parameters for the Goonjcustom cron.
+}
+
+/**
+ * Goonjcustom.induction_slot_booking_follow_up_cron API implementation.
+ *
+ * This function checks for unscheduled induction activities older than 7 days 
+ * and sends follow-up emails to the respective contacts if they haven't already 
+ * received one.
+ *
+ * @param array $params
+ *   Parameters passed to the API call.
+ *
+ * @return array API result descriptor
+ *
+ * @see civicrm_api3_create_success
+ * @see civicrm_api3_create_error
+ *
+ * @throws \CRM_Core_Exception
+ */
+function civicrm_api3_goonjcustom_induction_slot_booking_follow_up_cron($params) {
+    $returnValues = [];
+    \Civi::log()->info('Check for unscheduled induction activities');
+
+    // Configurable number of days to check for scheduling
+    $followUpDays = 7;
+
+    // Calculate the timestamp for 7 days ago from the current date
+    $followUpTimestamp = strtotime("-$followUpDays days");
+
+    $batchSize = 25;
+    $offset = 0;
+
+    try {
+        // Retrieve the email template for follow-up.
+        $template = MessageTemplate::get(FALSE)
+            ->addSelect('id', 'msg_subject')
+            ->addWhere('msg_title', 'LIKE', 'Induction_slot_booking_follow_up_email%')
+            ->execute()->single();
+
+        do {
+            // Retrieve a batch of unscheduled induction activities older than 7 days
+            $unscheduledInductionActivities = Activity::get(FALSE)
+                ->addSelect('id', 'source_contact_id', 'created_date')
+                ->addWhere('activity_type_id:name', '=', 'Induction')
+                ->addWhere('status_id:name', '=', 'To be scheduled')
+                ->addWhere('created_date', '<', date('Y-m-d H:i:s', $followUpTimestamp))
```

Comment:
If I create an unscheduled induction activity for November 28 and today’s date is November 5, will I receive a notification? Can you confirm this requirement and if yes then update the logic from < to <=

Reply:
As per requirement its older than 7 days not equal to 7 days